examples/data_preprocessing/MIMIC_III_dataset/[WIP]_prepare_MIMIC_III_data.ipynb

```diff
@@ -0,0 +1,1340 @@
+      "                  'reticular marking',\n",
+      "                  'reticulation',\n",
+      "                  'parenchymal scarring',\n",
+      "                  'peribronchial thickening',\n",
+      "                  'wall thickening',\n",
+      "                  'scar'],\n",
+      " 'pleural_effusion': ['pleural fluid', 'effusion'],\n",
+      " 'pleural_other': ['pleural thickening',\n",
+      "                   'fibrosis',\n",
+      "                   'fibrothorax',\n",
+      "                   'pleural scar',\n",
+      "                   'pleural parenchymal scar',\n",
+      "                   'pleuro-parenchymal scar',\n",
+      "                   'pleuro-pericardial scar'],\n",
+      " 'pneumonia': ['pneumonia', 'infection', 'infectious process', 'infectious'],\n",
+      " 'pneumothorax': ['pneumothorax', 'pneumothoraces'],\n",
+      " 'support_devices': ['pacer',\n",
+      "                     '_line_',\n",
+      "                     'lines',\n",
+      "                     'picc',\n",
+      "                     'tube',\n",
+      "                     'valve',\n",
+      "                     'catheter',\n",
+      "                     'pacemaker',\n",
+      "                     'hardware',\n",
+      "                     'arthroplast',\n",
+      "                     'marker',\n",
+      "                     'icd',\n",
+      "                     'defib',\n",
+      "                     'device',\n",
+      "                     'drain_',\n",
+      "                     'plate',\n",
+      "                     'screw',\n",
+      "                     'cannula',\n",
+      "                     'apparatus',\n",
+      "                     'coil',\n",
+      "                     'support',\n",
+      "                     'equipment',\n",
+      "                     'mediport']}\n"
+     ]
+    }
+   ],
+   "source": [
+    "pprint.pprint(mentions)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Knodle matrices\n",
+    "\n",
+    "Now we can proceed to create and save the $T$ and $Z$ matrices."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### T matrix\n",
+    "\n",
+    "The $T$ matrix is of shape (*number of rules* x *number of classes*) and contains the initial mapping from rules to the cor- responding classes. $T_{jk}=1$ if rule $j$ is indicative of class $k$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['T_matrix.lib']"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "T_df = pd.concat([pd.Series(v, name=k).astype(str) for k, v in mentions.items()], axis = 1)\n",
+    "T_df = pd.get_dummies(T_df.stack()).sum(level=1).T\n",
+    "\n",
+    "CheXpert_rules = list(T_df.index)\n",
```

Comment:
I think you don't use the `CheXpert_rules` anywhere in the code

Reply:
I'm using it now, since I corrected the Z-matrix (here `CheXpert_rules` is used for creating the columns).